### PR TITLE
Setpoint constraints from ESDL

### PR DESCRIPTION
**Pull request by jimrojerTNO**
This PR is to give the users the setpoint constraint functionality via ESDL.

For now this is done by setting a range constraint on an asset. This is not the most intuitive way as it can only be done via de ESDL browser, but it works for now and can later be extended upon.

I created a small testcase which duplicates the result for a case where the setpoints where provided in an hardcoded manner.

### CODE REVIEW

File: tests/test_setpoint_constraints.py
Comment (FJanssen-TNO) [inline tests/test_setpoint_constraints.py:1]: Do you want to keep all the old heatproblemsetpointconstraint checks or could forinstance results_4 be removed?

File: tests/models/unit_cases/case_3a_setpoint/src/run_3a.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    priority = 3
+
+    order = 2
+
+    def __init__(self, source):
+        self.source = source
+        self.function_nominal = 1e6
+
+    def function(self, optimization_problem, ensemble_member):
+        return optimization_problem.state(f"{self.source}.Heat_source")
```
Comment (FJanssen-TNO) [inline tests/models/unit_cases/case_3a_setpoint/src/run_3a.py:90]: Nor these goals are used. Or do you plan to remove the setpointconstraint checks from case_3a/src/run_3a.py ?

File: tests/models/unit_cases/case_3a_setpoint/src/run_3a.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+            state = f"{demand}.Heat_demand"
+
+            goals.append(TargetDemandGoal(state, target))
+
+        # for s in self.heat_network_components["source"]:
+        #     try:
+        #         target_flow_rate = parameters[f"{s}.target_flow_rate"]
+        #         goals.append(ConstantGeothermalSource(self, s, target_flow_rate))
+        #     except KeyError:
+        #         pass
```
Comment (FJanssen-TNO) [inline tests/models/unit_cases/case_3a_setpoint/src/run_3a.py:108]: same for this part of the code

File: src/rtctools_heat_network/esdl/esdl_additional_vars_mixin.py [hunk 3]
```diff
@@ -33,6 +73,7 @@ def temperature_regimes(self, carrier):
                 *self.heat_network_components.get("ates", []),
                 *self.heat_network_components.get("buffer", []),
                 *self.heat_network_components.get("heat_pump", []),
+                *self.heat_network_components.get("heat_pump_elec", []),
```
[the remark this answers is not in the document]
Comment (jimrojerTNO) [reply src/rtctools_heat_network/esdl/esdl_additional_vars_mixin.py:75]: good point

File: tests/models/unit_cases/case_3a_setpoint/src/run_3a.py (hunk 2 above)
Comment (jimrojerTNO) [reply tests/models/unit_cases/case_3a_setpoint/src/run_3a.py:108]: yup

File: tests/test_setpoint_constraints.py [hunk 4]
```diff
@@ -37,6 +37,22 @@ def test_setpoint_constraints(self):
         )
         results_4 = _heat_problem_4.extract_results()
 
+        import models.unit_cases.case_3a_setpoint.src.run_3a as run_3a
```
[the remark this answers is not in the document]
Comment (jimrojerTNO) [reply tests/test_setpoint_constraints.py:40]: done

File: tests/test_setpoint_constraints.py
Comment (jimrojerTNO) [reply tests/test_setpoint_constraints.py:1]: I understand the comment, but I would like to keep it in, as it allows to seperately check the ESDLAdditionalVarsMixin from the constraint functioning.

**Review by FJanssen-TNO — APPROVED** [review]

**Pull request merged**